src/utility.h

```diff
@@ -106,112 +92,92 @@ inline void lstrip(std::string &s, UNARY f = chkspace)
     s = s.substr(static_cast<size_t>(nonmatch_it - s.begin()));
 }
 
-//! remove certain trailing characters from the given string (by default
-//! whitespace characters)
+/// Remove certain trailing characters from the given string (by default: whitespace characters).
 template <typename UNARY = bool(char)>
-inline void rstrip(std::string &s, UNARY f = chkspace)
-{
-    while( !s.empty() && f(*s.rbegin()) )
+inline void rstrip(std::string &s, UNARY f = chkspace) {
+    while (!s.empty() && f(*s.rbegin()))
         s.pop_back();
 }
 
-//! remove certain leading and trailing characters from the given string (by
-//! default whitespace characters)
+/// Remove certain leading and trailing characters from the given string (by default: whitespace characters).
 template <typename UNARY = bool(char)>
-void strip(std::string &s, UNARY f = chkspace)
-{
+void strip(std::string &s, UNARY f = chkspace) {
     lstrip(s, f);
     rstrip(s, f);
 }
 
-//! checks whether the given string has the given prefix
-inline bool hasPrefix(const std::string &s, const std::string &prefix)
-{
+/// Checks whether the given string has the given prefix.
+inline bool hasPrefix(const std::string &s, const std::string &prefix) {
     return s.substr(0, prefix.length()) == prefix;
 }
 
-//! checks whether the given string has the given suffix
-inline bool hasSuffix(const std::string_view &s, const std::string &suffix)
-{
+/// Checks whether the given string has the given suffix.
+inline bool hasSuffix(const std::string_view &s, const std::string &suffix) {
     if (suffix.length() > s.length())
         return false;
 
     return s.substr(s.length() - suffix.length()) == suffix;
 }
 
-//! returns whether the given iterable sequence contains the given element \c val
+/// Returns whether the given iterable sequence contains the given element `val`.
 template <typename T, typename SEQ>
-bool matchesAny(const T &val, const SEQ &seq)
-{
-    for (const auto &e: seq)
-    {
+bool matchesAny(const T &val, const SEQ &seq) {
```

Comment:
C++ dos not provide a library function for this?

Reply:
There is `std::find()` but it requires start and end iterators and only works on containers. This here operates on sequences in a more general sense, anything that supports for loop iteration.